kube-client/src/lib.rs

```diff
@@ -455,7 +455,7 @@ mod test {
         // individual logs may or may not buffer
         let mut output = String::new();
         while let Some(line) = logs_stream.try_next().await? {
-            output.push_str(&String::from_utf8_lossy(&line));
+            output.push_str(&line);
         }
         assert_eq!(output, "kube 1\nkube 2\nkube 3\nkube 4\nkube 5\n");
```

Comment:
`LinesCodec` usage will mean you no longer get newline characters in a buffered set of lines.
A more accurate test could instead push to a vector and and `assert_eq!(output, ["kube 1", "kube 2", "kube 3", "kube 4", "kube 5"])` to ensure we get one event per line.